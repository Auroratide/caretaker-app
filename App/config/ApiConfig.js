
export const API_URL = 'http://localhost:8080';
// export const API_URL = 'http://10.0.2.2:8080'

export const defaultRequestHeaders = {
  'Accept': 'application/json'
};