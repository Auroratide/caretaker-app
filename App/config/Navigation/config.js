import HomeScreen from '../../views/Home/';
import CaretakerRolesMenu from '../../views/CaretakerRolesMenu/';
import Announcements from '../../views/Announcements/';
import UserProfile from '../../views/UserProfile/';
import CommunityAdmin from '../../views/CommunityAdmin/';

export const config = {
    initialRouteName: 'Home'
}

export const helperRoutes = {
    Home: {
      screen: HomeScreen,
      navigationOptions: {
        drawerLabel: 'Schedule'
      }
    },
    Announcements: {
      screen: Announcements,
      navigationOptions: {
        drawerLabel: 'Announcements'
      }
    },
    CaretakerRolesMenu: {
      screen: CaretakerRolesMenu,
      navigationOptions: {
        drawerLabel: 'Roles'
      }
    },
    UserProfile: {
      screen: UserProfile,
      navigationOptions: {
        drawerLabel: 'Profile'
      }
    }
}

export const adminRoutes = {
    ...helperRoutes,
    CommunityAdmin: {
      screen: CommunityAdmin,
      navigationOptions: {
        drawerLabel: 'Admin'
      }
    }
}
