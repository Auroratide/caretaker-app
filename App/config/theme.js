const PINK = '#FF7575';
const DARK_PINK = '#E66969';

const theme = {
  palette: {
    primaryColor: PINK,
    secondaryColor: DARK_PINK
  },
  toolbar: {
    container: {
      height: 50,
    },
  },
};

export default theme;
