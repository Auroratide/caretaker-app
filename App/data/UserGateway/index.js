export class Permissions {
    static FOCUS = 'focus';
    static ADMIN = 'admin';
    static HELPER = 'helper';
}

const createUserGateway = function() {
    let allUsers = null;

    return {
        currentUser: null,

        getAll() {
            if(allUsers === null) {
                //api call goes here
                allUsers = [
                    {name: 'Sarah', identifier: 'sarah@emailprovider.com', permission: Permissions.FOCUS, roles: [], phone:'123-456-7890'},
                    {name: 'Sarah', identifier: 's', permission: Permissions.FOCUS, roles: [], phone:'123-456-7890'},
                    {name: 'Caroline', identifier: 'caroline@woahdude.com', permission: Permissions.ADMIN, roles: ['driver', 'coordinator'], phone:'123-456-7890'},
                    {name: 'Jack', identifier: 'jack@coolwebsite.com', permission: Permissions.HELPER, roles: ['active friend']},
                    {name: 'Austin', identifier: 'austin@yeehaw.com', permission: Permissions.HELPER, roles: [], phone:'123-456-7890'}
                ]
            }
            return allUsers;
        },

        getName(identifier) {
            const user = allUsers.find((user) => user.identifier === identifier)
            return user ? user.name : 'TBD';
        },

        getRoles() {
            return this.currentUser.roles
        },

        signIn(email) {
            //make call to google oauth and store the results
            let results = {
                email: email
            };
            let userFromBackend = this.getAll().find((user) => user.identifier === results.email);
            if(typeof userFromBackend !== 'undefined') {
                this.currentUser = userFromBackend;
                return true;
            }
            return false;
        },

        isSelf(identifier) {
            return this.currentUser.identifier === identifier;
        },

        isAdmin() {
            return this.currentUser.permission === Permissions.ADMIN
                || this.currentUser.permission === Permissions.FOCUS;
        },

        getCurrentPermissions() {
            return this.currentUser.permission;
        },

        inviteUser(email) {
            //api call to invite the user
            alert('invited ' + email + '!');
        }
    }
}

export default createUserGateway();
