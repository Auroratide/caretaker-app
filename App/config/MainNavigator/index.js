import { StackNavigator } from 'react-navigation';
import Login from '../../views/Login/';
import Navigation from '../Navigation/';

export const LoginStack = StackNavigator({
    Login: { screen: Login },
    Home: { screen: Navigation }
}, {
       header: null,
       headerMode: 'none'
})
