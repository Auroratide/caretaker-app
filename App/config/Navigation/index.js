import React, { Component } from 'react';
import {DrawerNavigator, NavigationActions} from 'react-navigation';
import UserGateway, { Permissions } from '../../data/UserGateway/';
import {adminRoutes, helperRoutes, config} from './config.js';

export default class Navigation extends Component {
    constructor() {
        super();
        this.state = {
            userType: null
        }
    }

    componentDidMount() {
        const permissions = UserGateway.getCurrentPermissions();
        this.setState({userType: permissions});
    }

    _onLogout = () => {
        UserGateway.logout();
        const resetAction = NavigationActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Login'})]
        });
        this.props.navigation.dispatch(resetAction);
    }

    render() {
        const userType = this.state.userType;
        let result = null;
        const screenProps = {
            routes: routesFor[userType],
            logout: this._onLogout
        }
        if(userType) {
            const Navigator = DrawerNavigator(routesFor[userType], config);
            result = <Navigator screenProps={screenProps} />;
        }
        return result;
    }
}

const routesFor = {
    [Permissions.FOCUS]: adminRoutes,
    [Permissions.ADMIN]: adminRoutes,
    [Permissions.HELPER]: helperRoutes
}
