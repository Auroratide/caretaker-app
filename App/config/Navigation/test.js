import React, {Component} from 'react';
import renderer from 'react-test-renderer';
import {DrawerNavigator} from 'react-navigation';

import UserGateway, {Permissions} from '../../data/UserGateway/';
import Navigation from './index.js';
import {adminRoutes, helperRoutes, config} from './config.js';

describe('Navigation', () => {

    beforeEach(() => {
        DrawerNavigator.mockReset();
        DrawerNavigator.mockReturnValue(MockDrawerNavigator)
    })

    it('snapshot', () => {
        jest.spyOn(UserGateway, 'getCurrentPermissions')
            .mockReturnValue(Permissions.HELPER);
        expect(renderer.create(<Navigation />)).toMatchSnapshot();
    })

    it('should get the permissions from the user gateway', () => {
        jest.spyOn(UserGateway, 'getCurrentPermissions')
            .mockReturnValue(Permissions.HELPER);
        const navigation = renderer
            .create(<Navigation />)
            .getInstance();

        expect(navigation.state.userType).toBe(Permissions.HELPER);
    })

    it('should use the admin routes when the user is an admin', () => {
        jest.spyOn(UserGateway, 'getCurrentPermissions')
            .mockReturnValue(Permissions.ADMIN);
        const navigation = renderer
            .create(<Navigation />)
            .getInstance();

        expect(DrawerNavigator).toHaveBeenCalledWith(adminRoutes, config);
    })

    it('should use the admin routes when the user is a focus', () => {
        jest.spyOn(UserGateway, 'getCurrentPermissions')
            .mockReturnValue(Permissions.FOCUS);
        const navigation = renderer
            .create(<Navigation />)
            .getInstance();

        expect(DrawerNavigator).toHaveBeenCalledWith(adminRoutes, config);
    })

    it('should use the helper routes when the user is a helper', () => {
        jest.spyOn(UserGateway, 'getCurrentPermissions')
            .mockReturnValue(Permissions.HELPER);
            const navigation = renderer
                .create(<Navigation />)
                .getInstance();

        expect(DrawerNavigator).toHaveBeenCalledWith(helperRoutes, config);
    })

    it('should render null when the state is not set', () => {
        const navigation = renderer
            .create(<Navigation />)
            .getInstance();
        navigation.state.userType = null;

        expect(navigation.render()).toBe(null);
    })

})


class MockDrawerNavigator extends Component {
    render(){
        return (
            <div>Mock Drawer Navigator</div>
        );
    }
}
