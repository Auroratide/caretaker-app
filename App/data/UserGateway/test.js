import UserGateway from './index.js';

describe('UserGateway', () => {

    it('should return false when attempting to sign in with a non-existent user', () => {
        expect(UserGateway.login('TBD')).toBe(false);
    });

    it('should return true when sign in is successful', () => {
        expect(UserGateway.login('sarah@emailprovider.com')).toBe(true);
    });

    it('should assign TBD as a name when TBD is the identifier', () => {
        expect(UserGateway.getName('TBD')).toBe('TBD');
    })
});
