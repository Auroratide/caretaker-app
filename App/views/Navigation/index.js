import React, { Component } from 'react';

import {
  StyleSheet,
  Text,
  View
} from 'react-native';

import {AdminTabs, Tabs} from '../../config/MainNavigator';

import UserGateway, { Permissions } from '../../data/UserGateway';

export default class Navigation extends Component {
    constructor() {
        super();
        this.state = {
            userType: null
        }
    }

    componentDidMount() {
        const permissions = UserGateway.getCurrentPermissions();
        this.setState({userType: permissions});
    }

    render() {
        var tab = null
        switch (this.state.userType) {
            case Permissions.FOCUS:
                tab = <AdminTabs />;
                break;
            case Permissions.ADMIN:
                tab = <AdminTabs />;
                break;
            case Permissions.HELPER:
                tab = <Tabs />;
                break;
        }
        return tab;
    }
}

const screenFor = {
    [Permissions.FOCUS]: (<Tabs />),
    [Permissions.ADMIN]: (<Tabs />),
    [Permissions.HELPER]: (<Tabs />)
}

const styles = StyleSheet.create({
  button: {
    margin: 10,
    padding: 10
  },
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 24,
    textAlign: 'center',
    margin: 15,
  },
});