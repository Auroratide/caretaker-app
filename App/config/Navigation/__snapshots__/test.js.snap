// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot 1`] = `
<div>
  Mock Tab Navigator
</div>
`;
